src/tlo/methods/hsi_generic_first_appts.py

```diff
@@ -6,369 +6,179 @@
+        # further HSI will be needed for this person. In some cases,
+        # small bits of care are provided (e.g. a diagnosis, or the
+        # provision of inhaler).
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint(
+            {}
+        )
+
+    def _diagnosis_fn(self, tests, use_dict: bool = False, report_tried: bool = False) -> Any:
+        """
+        Passed to modules when determining HSI_Events to be scheduled based on
+        this generic appointment. Intended as the diagnosis_fn argument to the
+        Module.do_at_generic_{non_}_emergency.
+
+        Class-level definition avoids the need to redefine this method each time
+        the .apply() method is called.
+
+        :param tests: The name of the test(s) to run via the diagnosis manager.
+        :param use_dict_for_single: If True, the return type will be a dictionary
+        even if only one test was requested.
+        :param report_dxtest_tried: Report if a test was attempted but could not
+        be carried out due to EG lack of consumables, etc.
+        :returns: Test results as dictionary key/value pairs.
+        """
+        return self.healthcare_system.dx_manager.run_dx_test(
+            tests,
+            hsi_event=self,
+            use_dict_for_single=use_dict,
+            report_dxtest_tried=report_tried,
+        )
+
+    def _do_on_generic_first_appt(self, squeeze_factor: float = 0.) -> None:
+        """
+        """
+        # Make top-level reads of information, to avoid repeat accesses.
+        modules: OrderedDict[str, "Module"] = self.sim.modules
+        schedule_hsi = self.healthcare_system.schedule_hsi_event
+        symptoms = modules["SymptomManager"].has_what(self.target)
 
-class HSI_GenericNonEmergencyFirstAppt(HSI_Event, IndividualScopeEventMixin):
-    """This is a Health System Interaction Event that represents the first interaction with the health system following
-     the onset of non-emergency symptom(s). This is the HSI that is generated by the HealthSeekingBehaviour module. By
-     default, it occurs at level '0' but it could occur also at other levels."""
+        # Dynamically create immutable container with the target's details stored.
+        # This will avoid repeat DataFrame reads when we call the module-level functions.
+        df = self.sim.population.props
+        patient_details = namedtuple("PatientDetails", df.columns)(*df.loc[self.target])
+
+        proposed_df_updates = {}
+
+        for module in modules.values():
+            event_info, df_updates = getattr(module, self.MODULE_METHOD_ON_APPLY)(
+                patient_id=self.target,
+                patient_details=patient_details,
+                symptoms=symptoms,
+                diagnosis_fn=self._diagnosis_fn,
+                consumables_checker=self.get_consumables,
+                facility_level=self.ACCEPTED_FACILITY_LEVEL,
+                treatment_id=self.TREATMENT_ID,
+                random_state=self.module.rng,
```

Comment:
Looks like the random number generator is only used by one module downstream (`Labour`). It might be worth checking either if (i) the random sampling used there is actually required (seems to be randomly assigning facility level) and (ii) if it might not make more sense to use the module specific `RandomState` instance under the `rng` attribute as that is generally the pattern used in other methods of  `Module` instances.

Reply:
Yeah it is only used by the `labour` module, however switching to the `labour` module's `rng` attribute then causes the deterministic simulation test to fail (clearly the two random states generate a different random choice for the facility level at some point).

Reply:
Hmm - I'm surprised that test fails as the module specific random number generator instances are also also deterministically seeded. I wonder if there another occurence of the dreaded non-deterministic ordering of iterating over sets coming in to play 🤔. I will look in to this more!